doc/guides/modules.rst

```diff
@@ -421,6 +421,12 @@ CMake variable ``ZEPHYR_<MODULE_NAME>_MODULE_DIR`` and the variable
 ``ZEPHYR_<MODULE_NAME>_CMAKE_DIR`` holds the location of the directory
 containing the module's :file:`CMakeLists.txt` file.
 
+.. note::
+   All symbols in module names will be replaced with an underscore, ``_``, and
+   converted to upper characters when used for CMake and Kconfig variables.
```

Comment:
This sentence could be rephrased, as right now it is a bit unclear.

Reply:
How about:
>  The CMake and Kconfig variables will be the module name converted to upper characters, and all symbols will be replaced with an underscore, ``_``.

is that clearer ?

Reply:
Maybe:
> When used for CMake and Kconfig variables, all letters in module names are
converted to uppercase and all non-alphanumeric characters are converted
to underscores (``_``).

Reply:
done.